merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MissionModelGenerator.java

```diff
@@ -461,49 +523,120 @@ public JavaFile generateActivityTypes(final MissionModelRecord missionModel) {
                     .addMember("value", "$S", MissionModelProcessor.class.getCanonicalName())
                     .build())
             .addModifiers(Modifier.PUBLIC, Modifier.FINAL)
-            .addField(
-                FieldSpec
-                    .builder(
+            .addSuperinterface(
+                ParameterizedTypeName.get(
+                    ClassName.get(Scoping.class),
+                    missionModel.getTypesName(),
+                    ClassName.get(missionModel.topLevelModel)))
+            .addFields(
+                missionModel.activityTypes
+                    .stream()
+                    .map(activityType -> FieldSpec
```

Comment:
I see now why you were, in all seriousness, looking up the maximum number of fields allowed in a Java class 😆

Reply:
Right??